app/actions/remote/thread.ts

```diff
@@ -361,140 +300,179 @@ async function fetchBatchThreads(
         if (__DEV__) {
             throw error;
         }
-
-        return {error, data};
+        return {error};
     }
 
-    return {error: false, data};
-}
-
-export async function fetchNewThreads(
-    serverUrl: string,
-    teamId: string,
-    prepareRecordsOnly = false,
-): Promise<{error: unknown; models?: Model[]}> {
-    const options: FetchThreadsOptions = {
-        unread: false,
-        deleted: true,
-        perPage: 60,
-    };
+    return {error: false, threads: threadsData};
+};
 
+export const syncTeamThreads = async (serverUrl: string, teamId: string, prepareRecordsOnly = false) => {
     const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
-
     if (!operator) {
         return {error: `${serverUrl} database not found`};
     }
 
-    const newestThread = await getNewestThreadInTeam(operator.database, teamId, false);
-    options.since = newestThread ? newestThread.lastReplyAt : 0;
-
-    let response: {
-        error: unknown;
-        data?: Thread[];
-    } = {
-        error: undefined,
-        data: [],
-    };
-
-    let loadedInGlobalThreads = true;
+    try {
+        const syncData = await getTeamThreadsSyncData(operator.database, teamId);
+        const syncDataUpdate = {
+            id: teamId,
+        } as TeamThreadsSync;
+
+        const threads: Thread[] = [];
+
+        /**
+         * If Syncing for the first time,
+         *     - Get all unread threads to show the right badges
+         *     - Get latest threads to show by default in the global threads screen
+         * Else
+         *     - Get all threads since last sync
+         */
+        if (!syncData || !syncData?.latest) {
+            const [allUnreadThreads, latestThreads] = await Promise.all([
+                fetchThreads(
+                    serverUrl,
+                    teamId,
+                    {unread: true},
+                    Direction.Down,
+                ),
+                fetchThreads(
+                    serverUrl,
+                    teamId,
+                    {},
+                    undefined,
+                    1,
+                ),
```

Comment:
I do not quite get the logic here, so I'll defer to @koox00

Reply:
LGTM